core/src/main/java/com/amplifyframework/auth/AWSCognitoAuthSession.kt

```diff
@@ -0,0 +1,46 @@
+package com.amplifyframework.auth
+
+import com.amplifyframework.auth.cognito.helpers.JWTParser
+import com.amplifyframework.auth.exceptions.UnknownException
+import com.amplifyframework.auth.result.AuthSessionResult
+
+interface AWSCognitoAuthSession {
+    val isSignedIn: Boolean
+    val identityIdResult: AuthSessionResult<String>
+    val awsCredentialsResult: AuthSessionResult<AWSCredentials>
+    val userSubResult: AuthSessionResult<String>
+    val userPoolTokensResult: AuthSessionResult<AWSCognitoUserPoolTokens>
+
+    companion object {
```

Comment:
This companion object is implementation detail and is used only in auth plugin. I believe this should be in auth plugin.

Reply:
I noticed that those methods are only just used in one function. Moved them there.